src/main/java/com/cloudbees/plugins/credentials/GlobalCredentialsConfiguration.java

```diff
@@ -221,6 +222,7 @@ public String getDisplayName() {
      * Security related configurations.
      */
     @Extension
+    @Symbol("globalCredentialsConfiguration")
```

Comment:
shouldn't it just be `globalCredentials` (unless there's a clash)?
```suggestion
    @Symbol("globalCredentials")
```

what does it look like in the export currently?

Reply:
Like this:

```
globalCredentialsCategory:
  cyberArkGlobalConfiguration:
    allowCustomAppIds: false
```

Before:

```
? ''
: cyberArkGlobalConfiguration:
    allowCustomAppIds: false
```

Reply:
I don't really understand why this isn't in the 'credentials' category (but possibly too late).

but the symbol shouldn't have a suffix of category as it's implied by the extension point.

Do you have a symbol added for `cyberArkGlobalConfiguration`? normally the `GlobalConfiguration` would be stripped out by jcasc unless you actually put it in your symbol name.

I would expect the yaml to be:

```yaml
globalCredentials:
  cyberArk:
    allowCustomAppIds: false
```

Reply:
> I don't really understand why this isn't in the 'credentials' category (but possibly too late).

Because it's not defining _credentials_, it's defining credential stores. It's under `/jenkins/configureCredentials/` - I didn't even know about this feature! Just learnt it today.

Reply:
> Do you have a symbol added for cyberArkGlobalConfiguration? normally the GlobalConfiguration would be stripped out by jcasc unless you actually put it in your symbol name.

Yeah, I have a symbol, but it doesn't really have any impact on this, I can set the symbol there to whatever and the root element is still empty (because it's the category!) - I'm not saying this makes sense, just stating the current behavior.

So it looks like `GlobalCategory` extensions behave differently when they are in the `GlobalCredentialsConfiguration.Category` (which is the global configuration implementing [`Descriptor#getCategory`](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/Descriptor.java#L831), which from a descriptor configuration stand point kind of makes sense as [`GlobalCredentialsConfiguration#configure`](https://github.com/jenkinsci/credentials-plugin/blob/00173e541b29a5c742954ecd69f509d35ca92208/src/main/java/com/cloudbees/plugins/credentials/GlobalCredentialsConfiguration.java#L163) is acting as a root configurator element for all `GlobalConfiguration` under the category.

Reply:
I applied a patch against discovering the empty category.
https://github.com/jenkinsci/configuration-as-code-plugin/blob/02ca4728571a26465339742f532c9e50fe629b07/plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/GlobalConfigurationCategoryConfigurator.java#L125-L129

Reply:
We also have code that removes `Category` 
https://github.com/jenkinsci/configuration-as-code-plugin/blob/02ca4728571a26465339742f532c9e50fe629b07/plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/GlobalConfigurationCategoryConfigurator.java#L58-L62

Reply:
This should allow removing that patch? The second example is for guessing the name and is the root cause of this

Reply:
Well we could remove it if it were not for older versions of the plugin.

Does it even work with this patch @amuniz when exporting it?

Reply:
https://github.com/jenkinsci/credentials-plugin/pull/144#discussion_r407452543

Reply:
> Please use a recently new version of JCasC

I'm using the latest release (1.38) in my local tests.

> Does it even work with this patch @amuniz when exporting it?

Yes, it does.

Reply:
I guess it works because the child is discovered 🤔

Reply:
I just added a test to demonstrate the behavior, it might be useful to understand the issue @jetersen